gtfs/importers/gtfs_feed_importer.py

```diff
@@ -315,10 +367,52 @@ def _get_default_agency_id(self):
         if len(agency_ids) != 1:
             raise GTFSFeedImporterError(
                 f"Something fishy going on! Trying to use default agency but there "
-                f"are {len(agency_ids)}) agencies in the feed."
+                f"are {len(agency_ids)} agencies in the feed."
             )
         return next(iter(agency_ids))
 
+    def _form_translation_list(self, gtfs_data):
+        iterable_data = (
+            gtfs_data if isinstance(gtfs_data, list) else gtfs_data.itertuples()
+        )
+        translations_list = []
+        for _num_of_processed, gtfs_row in enumerate(iterable_data, 1):
+            translation_fields = {}
+
+            for model_field_name, gtfs_field in self.TRANSLATION_MAPPING.items():
+                gtfs_value = getattr(gtfs_row, gtfs_field, None)
+                translation_fields[model_field_name] = self._convert_str(gtfs_value)
+            translations_list.append(translation_fields)
+
+        return translations_list
+
+    def _add_translations(self, model, translations_list):
+        plural_name = model._meta.verbose_name_plural
+        translations_for_model = [
+            t for t in translations_list if t.get("table_name") == plural_name
+        ]
+        num_of_translations = len(translations_for_model)
+
+        if num_of_translations == 0:
+            return
+        else:
+            self.logger.info(
+                f"Adding {num_of_translations} translations for {plural_name}..."
+            )
+
+        for trans in translations_for_model:
+            for model_field_name, gtfs_field in self.FIELD_MAPPING[model].items():
+                if trans.get("field_name") == gtfs_field:
+                    trans["field_name"] = model_field_name
+
+        obj_list = model.objects.all()
+        for obj in obj_list:
+            for trans in translations_for_model:
+                if trans.get("record_id") == obj.source_id:
+                    obj.set_current_language(trans.get("language"))
+                    setattr(obj, trans.get("field_name"), trans.get("translation"))
+            obj.save()
```

Comment:
While premature optimization is the root of all evil and all that, I'm still a bit worried that because of the nested loopings this will choke pretty soon when there is more data to be imported. I think the goal should be to loop over `translations_for_model` only once and update the objects while doing that. It would probably make this less complicated also.

Of course, it might very well be that I'm missing something here 😄

Reply:
This definitely needs some improvements. I have no idea why I wrote most of the code here like a caveman 😅